src/redis_index.c

```diff
@@ -203,6 +202,8 @@ RedisSearchCtx *NewSearchCtx(RedisModuleCtx *ctx, RedisModuleString *indexName,
 void RedisSearchCtx_UnlockSpec(RedisSearchCtx *sctx) {
   if (sctx->flags == RS_CTX_UNSET) {
     return;
+  } else if (sctx->flags == RS_CTX_READWRITE) {
+      IndexSpec_UpdateVersion(sctx->spec);
```

Comment:
this means that every locking for writing counts as an edit, even from the GC

Reply:
We can consider to add 'RedisSearchCtx_UnlockSpecAndUpdate' function in the touch up phase